src/test/daml/Daml/Finance/Instrument/Option/Test/Util.daml

```diff
@@ -0,0 +1,85 @@
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Instrument.Option.Test.Util where
+
+import DA.List (head, sort)
+import DA.Map qualified as M (empty, fromList)
+import DA.Set qualified as S (empty, singleton)
+import Daml.Finance.Claims.Lifecycle.Rule (Rule(..))
+import Daml.Finance.Instrument.Option.European.Instrument qualified as EuropeanOption (Instrument(..))
+import Daml.Finance.Interface.Instrument.Option.European.Types (OptionTypeEnum)
+import Daml.Finance.Interface.Lifecycle.Effect qualified as Effect (I, GetView(..))
+import Daml.Finance.Interface.Lifecycle.Observable.NumericObservable qualified as NumericObservable (I)
+import Daml.Finance.Interface.Lifecycle.Rule.Lifecycle qualified as Lifecycle (I, Evolve(..))
+import Daml.Finance.Interface.Types.Common.Types (Id(..), InstrumentKey, InstrumentQuantity, Parties)
+import Daml.Finance.Test.Util.Instrument (createReference)
+import Daml.Finance.Test.Util.Time (createClockUpdateEvent)
+import Daml.Script
+
+-- | Originate a European option.
+originateEuropeanOption : Party -> Party -> Text -> Text -> [(Text, Parties)] -> Time ->
+  Date -> OptionTypeEnum -> Decimal -> InstrumentKey -> Text -> Bool -> Script InstrumentKey
+originateEuropeanOption depository issuer label description observers lastEventTimestamp
+  expiryDate optionType strike currency referenceAssetId ownerReceives = do
+    cid <- toInterfaceContractId <$> submitMulti [depository, issuer] [] do
+      createCmd EuropeanOption.Instrument with
+        depository; issuer; id = Id label; version = "0"; description
+        observers = M.fromList observers; lastEventTimestamp
+        ownerReceives; optionType; strike; expiryDate
+        referenceAssetId; currency
+    createReference cid depository issuer observers
+
+-- | Lifecycle the instrument as of this date.
+lifecycleInstrument : [Party] -> Date -> InstrumentKey -> Party ->
+  [ContractId NumericObservable.I] -> Script (Optional InstrumentKey, [ContractId Effect.I])
+lifecycleInstrument readAs today instrument issuer observableCids = do
+  -- create clock update event
+  clockEventCid <- createClockUpdateEvent (S.singleton issuer) today S.empty
+
+  -- Create a lifecycle rule
+  lifecycleRuleCid <- toInterfaceContractId @Lifecycle.I <$> submit issuer do
+    createCmd Rule with
+      providers = S.singleton issuer
+      observers = M.empty
+      lifecycler = issuer
+      id = Id "LifecycleRule"
+      description = "Rule to lifecycle an option instrument"
+
+  -- Try to lifecycle instrument
+  (lifecycleCid, effectCids) <- submitMulti [issuer] readAs do
+    exerciseCmd lifecycleRuleCid Lifecycle.Evolve with
+      eventCid = clockEventCid
+      observableCids
+      instrument
+
+  pure (lifecycleCid, effectCids)
+
+-- | Verify a that there are no lifecycle effects of the instrument on this date.
+verifyNoLifecycleEffects : [Party] -> Date -> InstrumentKey -> Party ->
```

Comment:
If these functions are the same that are used for bonds and swaps, we should consider putting then in a common place.

Reply:
Good point, I have opened a separate issue for that: https://github.com/digital-asset/daml-finance/issues/599